### PR TITLE
feat: apply new colors in settings screen- ACC-225

### PR DESCRIPTION
**Pull request by BrzozowskiPawel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-225" title="ACC-225" target="_blank">SubtaskACC-225</a>  [iOS] Apply new colors in Settings Screen
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Updated color for the new Figma design in the settings screen.

### Issues

The colors and design were outdated.

### Solutions

Changing the properties to match the new design.

#### Test Coverage (Optional)

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Settings/ChangingDetails/ConfirmPhoneViewController.swift [hunk 1]
```diff
@@ -152,10 +152,8 @@ final class ConfirmPhoneViewController: SettingsBaseTableViewController {
             let cell = tableView.dequeueReusableCell(withIdentifier: SettingsButtonCell.zm_reuseIdentifier, for: indexPath) as! SettingsButtonCell
             cell.titleText = "self.settings.account_section.phone_number.change.verify.resend".localized
             if resendEnabled {
-                cell.titleColor = .white
                 cell.selectionStyle = .default
             } else {
-                cell.titleColor = UIColor(white: 1, alpha: 0.4)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/ChangingDetails/ConfirmPhoneViewController.swift:158]: You can simplify this:
cell.selectionStyle = resendEnabled ? .default : .none

File: Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientColorVariantProtocol.swift [hunk 2]
```diff
@@ -51,13 +46,10 @@ extension ClientColorVariantProtocol where Self: UIViewController {
...
-        switch variant {
-        case .none:
+        if variant == .none {
             view.backgroundColor = .clear
-        case .dark?:
-            view.backgroundColor = .black
-        case .light?:
-            view.backgroundColor = .white
+        } else {
+            view.backgroundColor = SemanticColors.Background.settingsView
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientColorVariantProtocol.swift:52]: view.backgroundColor = (variant == .none) ? .clear : SemanticColors.Background.settingsView

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 3]
```diff
@@ -91,41 +91,20 @@ class SettingsTableCell: UITableViewCell, SettingsCellType {
...
 
-    var titleText: String = "" {
+    var isTransparent: Bool = false {
         didSet {
-            cellNameLabel.text = titleText
+            backgroundColor = isTransparent ? .clear : SemanticColors.Background.settingsTableCell
         }
     }
-
+
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:107]: Swift lint issue

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 4]
```diff
@@ -180,7 +159,8 @@ class SettingsTableCell: UITableViewCell, SettingsCellType {
     var icon: StyleKitIcon? {
         didSet {
             if let icon = icon {
-                iconImageView.setIcon(icon, size: .tiny, color: UIColor.white)
+                iconImageView.tintColor = SemanticColors.LabelsColor.textLabelSettingsCell
+                iconImageView.setTemplateIcon(icon, size: .tiny)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:163]: Why not `iconImageView.setIcon(icon, size: .tiny, color: SemanticColors.LabelsColor.textLabelSettingsCell)` ?

Comment (BrzozowskiPawel) [reply Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:163]: Agis suggested this approach as when we use the setIcon() it's not auto updating when the user changes the theme.

**Comment by KaterinaWire** [comment]
@BrzozowskiPawel I saw in `ConversationOptionsViewControllerTests` that only a few cells have separators? Is it correct?

**Comment by KaterinaWire** [comment]
@BrzozowskiPawel @agisilaos  Shouldn't the cells in `ClientListViewControllerTests` have a white background in the light mode?

**Comment by BrzozowskiPawel** [comment]
@KaterinaWire 
1. Yes there was an issue. Other cells should have also different UI. This was a screen that is entered thru messages. Besides of that somehow I was not able to have this sharing link on the simulator but I am fixing it already.
2. Yes this is also a valid case. I could not find this cell inside the fimga's preferences or conversation but I have finally found it and applied new design.

**Review by agisilaos — CHANGES_REQUESTED** [review]
I left some comments regarding to how we set the color for icons. Let's fix those first before making the PR ready to be merged.

**Review by agisilaos — APPROVED** [review]

File: Wire-iOS Tests/DeviceManagement/ClientListViewControllerTests.swift [hunk 5]
```diff
@@ -69,7 +69,7 @@ final class ClientListViewControllerTests: XCTestCase, CoreDataFixtureTestHelper
 
         sut = ClientListViewController(clientsList: clientsList,
                                        selfClient: selfClient,
-                                       credentials: nil,
+                                        credentials: nil,
```
Comment (KaterinaWire) [inline Wire-iOS Tests/DeviceManagement/ClientListViewControllerTests.swift:72]: Do we need these changes?

File: Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/SettingsClientViewController.swift [hunk 6]
```diff
@@ -262,15 +261,15 @@ final class SettingsClientViewController: UIViewController,
             if let cell = tableView.dequeueReusableCell(withIdentifier: type(of: self).resetCellReuseIdentifier, for: indexPath) as? SettingsTableCell {
                 cell.titleText = NSLocalizedString("profile.devices.detail.reset_session.title", comment: "")
                 cell.accessibilityIdentifier = "reset session"
-                cell.variant = self.variant
+                cell.isTransparent = variant == .none ? true : false
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/SettingsClientViewController.swift:264]:
```suggestion
                cell.isTransparent = variant == .none
```

File: Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/SettingsClientViewController.swift [hunk 7]
```diff
@@ -262,15 +261,15 @@ final class SettingsClientViewController: UIViewController,
...
                 return cell
             }
 
         case .removeDevice:
             if let cell = tableView.dequeueReusableCell(withIdentifier: type(of: self).deleteCellReuseIdentifier, for: indexPath) as? SettingsTableCell {
                 cell.titleText = NSLocalizedString("self.settings.account_details.remove_device.title", comment: "")
                 cell.accessibilityIdentifier = "remove device"
-                cell.variant = self.variant
+                cell.isTransparent = variant == .none ? true : false
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/SettingsClientViewController.swift:272]:
```suggestion
                cell.isTransparent = (variant == .none)
```

File: Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift [hunk 8]
```diff
@@ -189,23 +168,7 @@ class SettingsTableCell: UITableViewCell, SettingsCellType {
         }
     }
 
-    var isFirst: Bool = false {
-        didSet {
-            topSeparatorLine.isHidden = !isFirst
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/SettingsCell.swift:194]: Why don't we need this logic?

**Comment by BrzozowskiPawel** [comment]
Requested changes applied.

File: Wire-iOS Tests/ChangeEmailViewControllerTests.swift [hunk 9]
```diff
@@ -26,17 +26,22 @@ final class ChangeEmailViewControllerTests: ZMSnapshotTestCase {
...
 
-        viewController.overrideUserInterfaceStyle = .dark
-        viewController.view.backgroundColor = .black
+        let viewController = sut.wrapInNavigationController(navigationControllerClass: NavigationController.self)
+
+//        let viewController = sut.wrapInNavigationController(navigationControllerClass: SettingsStyleNavigationController.self)
+//
+//        viewController.overrideUserInterfaceStyle = .dark
+//        viewController.view.backgroundColor = .black
+
```
Comment (KaterinaWire) [inline Wire-iOS Tests/ChangeEmailViewControllerTests.swift:36]: Swift lint issue and commented code

File: Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift [hunk 10]
```diff
@@ -351,7 +347,11 @@ final class ClientListViewController: UIViewController,
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         if let cell = tableView.dequeueReusableCell(withIdentifier: ClientTableViewCell.zm_reuseIdentifier, for: indexPath) as? ClientTableViewCell {
             cell.selectionStyle = .none
-            cell.accessoryType = self.detailedView ? .disclosureIndicator : .none
+            let image = UIImage(systemName: "chevron.right")
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift:350]: Why do we need this?

Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Settings/DeviceManagement/ClientListViewController.swift:350]: Don't we have a helper method `showDisclosureIndicator()` ?

File: Wire-iOS/Sources/Helpers/SemanticColors.swift [hunk 11]
```diff
@@ -47,6 +53,10 @@ public enum SemanticColors {
         static let textCellTitle = UIColor(light: Asset.black, dark: Asset.white)
         static let textCellSubtitle = UIColor(light: Asset.gray90, dark: Asset.white)
         static let textNoResults = UIColor(light: Asset.black, dark: Asset.gray20)
+        static let textSettingsPasswordPlaceholder = UIColor(light: Asset.gray70, dark: Asset.gray60)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Helpers/SemanticColors.swift:56]: We will use the placeholder color not only for settings. I would use the same as for SearchBar: `textInputViewPlaceholder`

File: Wire-iOS/Sources/Helpers/UIView+Borders.swift [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+public extension UIView {
+    func addTopBorder(color: UIColor) {
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/Helpers/UIView+Borders.swift:22]: I would create a generic method addBorder(...) with a parameter indicating what exactly the board is. As a result, we don't need to create a separate method for each border and duplicate code. I think we can define a default color as `SemanticColors.View.backgroundSeparatorCell` with a possibility to set any other color.

File: Wire-iOS/Sources/UserInterface/Availability/TitleView.swift [hunk 13]
```diff
@@ -109,7 +109,9 @@ class TitleView: UIView, DynamicTypeCapable {
 extension NSTextAttachment {
     static func downArrow(color: UIColor) -> NSTextAttachment {
         let attachment = NSTextAttachment()
-        attachment.image = StyleKitIcon.downArrow.makeImage(size: 8, color: color)
+        attachment.image = StyleKitIcon.downArrow.makeImage(
+            size: 8,
+            color: SemanticColors.Icon.foregroundPlainDownArrow).withRenderingMode(.alwaysTemplate)
```
Comment (KaterinaWire) [inline Wire-iOS/Sources/UserInterface/Availability/TitleView.swift:114]: Why do we need `withRenderingMode(.alwaysTemplate) `?

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**